### DISCUSSION TITLE
Move HelmChart/HelmRepository to helm-controller?

### DISCUSSION THREAD

**Discussion by seaneagan** [fluxcd/flux2#233] [Q&A] [answered]
HelmChart/HelmRepository only seem relevant in the context of HelmRelease. So if someone is not interested in using helm, and say only kustomize, if these CRDs/controllers were moved out of source-controller, they could avoid needing to deploy them. Conversely if someone is only interested in helm, and doesn't need to source any charts from git, they could avoid needing to deploy the source-controller (GitRepository CRD/controller).

Any code that needs to be reused between git and helm sources could go in the fluxcd/toolkit libs. In this case I believe the source-controller would only have GitRepository, so it could potentially make sense to call it the git-controller. Any future source types could be placed under their own controllers e.g. s3-controller or oci-controller.

**Comment by seaneagan** [comment]
We'd probably want to rename `gotk <action> source<s> <git/helm>` > `gotk <action> <git|helm>repositor<y/ies>`.

**Comment by stefanprodan** [comment, marked as answer]
I don't see a problem with users installing the toolkit CRDs even if they don't use them all. When we started the toolkit one goal was to separate the source operations from the reconcilers, so no, moving the helm repo to helm-controller is not an option in my mind.

**Reply by hiddeco** [reply]
If the abilities of the source-controller are a concern because the user only wants to support a set of sources in their cluster, we can always make the changes required to selectively start the reconcilers based on some configuration.
